src/dune_pkg/rev_store.ml

```diff
@@ -307,7 +307,8 @@ module At_rev = struct
        1. using libgit or ocamlgit
        2. possibly using [$ git archive] *)
     File.Set.filter files_at_rev ~f:(fun (file : File.t) ->
-      Path.Local.is_descendant file.path ~of_:path)
+      (not (Path.Local.equal file.path path))
```

Comment:
Could you add a comment here why this check exists? Otherwise its hard to understand what it is for.